redux/slices/traineeActionsSlice.ts

```diff
@@ -23,7 +25,19 @@ export const fetchTraineeActionsData = createAsyncThunk(
     const actionsService = new TraineeActionsService();
     const response: AxiosResponse<TraineeAction[]> =
       await actionsService.getIncompleteTraineeActions();
-    return response.data;
+    const rawData = response.data;
+
+    const finalData = rawData.map(action => ({
+      ...action,
+      availableFrom: new Date(action.availableFrom),
+      dueBy: new Date(action.dueBy),
+      completed: action.completed ? new Date(action.completed) : null,
+      tisReferenceInfo: {
+        ...action.tisReferenceInfo,
+        type: action.tisReferenceInfo.type
```

Comment:
Is there a reason to pull 'type' out explicitly? `tisReferenceInfo` will have `id` and `type` fields, as used in various places in the FE code. If there is a concern about the fields changing in `tisReferenceInfo` wouldn't you want to be explicit about `id` as well?

Reply:
Oops! No. Good spot. Just leftover code from when I was console-logging the hell out of things back in early days.
(If you mark PR as 'changes required', I'll wait for all your comments before updating.)